### PR TITLE
ex-3 Array hash

### PR DESCRIPTION
**Pull request by bansalsumit**

### CODE REVIEW

File: ruby/array_hash/main.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Create a method for Array that returns a hash having 'key' as length of
+# the element and value as an array of all the elements of that length.
+# Make use of Array#each.
+# Returned Hash should be sorted by key.
+
+require_relative "lib/array_hash"
+
+list = List.new
+p list.element_by_length_and_sort_by_key(gets.chop.split(" "))
```
Comment (ShubhamGupta) [inline ruby/array_hash/main.rb]: Can a `Hash` be sorted?

File: ruby/array_hash/lib/array_hash.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class List
```
Comment (ShubhamGupta) [inline ruby/array_hash/lib/array_hash.rb]: The question asks you to specifically monkey-patch array class. Please do that.

File: ruby/array_hash/lib/array_hash.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+  def element_by_length_and_sort_by_key(input_list)
+    input_list.each.group_by { |element| element.length }.sort.to_h
```
Comment (ShubhamGupta) [inline ruby/array_hash/lib/array_hash.rb]: What is the complexity of this line? Improve it.

**Review by ShubhamGupta — CHANGES_REQUESTED** [review]

File: ruby/array_hash/main.rb (hunk 1 above)
Comment (bansalsumit) [reply ruby/array_hash/main.rb]: yes sir.

**Pull request merged**